hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NNStorageRetentionManager.java

```diff
@@ -192,7 +191,7 @@ private long getImageTxIdToRetain(
       return 0L;
     }
 
-    TreeSet<Long> imageTxIds = Sets.newTreeSet(Collections.reverseOrder());
+    TreeSet<Long> imageTxIds = new TreeSet<>(Collections.reverseOrder());
```

Comment:
Note: It is recommended to use the TreeSet constructor directly.
https://github.com/google/guava/blob/v30.1.1/guava/src/com/google/common/collect/Sets.java#L399-L403

Reply:
That's correct. Hence, I have also created this follow-up Jira HADOOP-17726 for the same so that empty set constructors can be directly created.